napari/layers/utils/plane.py

```diff
@@ -0,0 +1,166 @@
+from typing import Tuple
+
+import numpy as np
+from pydantic import validator
+
+from ...utils.events import EventedModel, SelectableEventedList
+from ...utils.geometry import intersect_line_with_plane_3d
+from ...utils.translations import trans
+
+
+class Plane(EventedModel):
+    """Defines a plane in 3D (with optional thickness) and provides utility methods
+    and events to handle it.
+
+    A plane is defined by a position, a normal vector and a thickness value, and can
+    be enabled or disabled.
+
+    Attributes
+    ----------
+    position : 3-tuple
+        A 3D position on the plane, defined in data coordinates.
+    normal : 3-tuple
+        A 3D unit vector normal to the plane, defined in data coordinates.
+    thickness : float
+        Thickness of the slice.
+    enabled : bool
+        Whether the plane is considered enabled.
+    """
+
+    position: Tuple[float, float, float] = (0, 0, 0)
+    normal: Tuple[float, float, float] = (1, 0, 0)
+    enabled: bool = False
+    thickness: float = 10.0
```

Comment:
Should we make default thickness be 0? 10 seems a little random.

Reply:
Oops, copy-paste mistake!